src/payment/strategies/adyenv3/adyenv3-payment-strategy.ts

```diff
@@ -0,0 +1,361 @@
+import isVaultedInstrument from '../../is-vaulted-instrument';
+import Payment, { HostedInstrument } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { isCardState, AdyenAction, AdyenActionType, AdyenAdditionalAction, AdyenAdditionalActionState, AdyenClient, AdyenComponent, AdyenComponentType, AdyenError, AdyenPaymentMethodType, AdyenPlaceholderData, CardStateErrors } from './adyenv3';
+import AdyenV3PaymentInitializeOptions from './adyenv3-initialize-options';
+import AdyenV3ScriptLoader from './adyenv3-script-loader';
+
+export default class Adyenv3PaymentStrategy implements PaymentStrategy {
+    private _adyenClient?: AdyenClient;
+    private _cardVerificationComponent?: AdyenComponent;
+    private _paymentComponent?: AdyenComponent;
+    private _paymentInitializeOptions?: AdyenV3PaymentInitializeOptions;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _scriptLoader: AdyenV3ScriptLoader,
+        private _locale: string
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+
+        const { adyenv3 } = options;
+
+        if (!adyenv3) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.adyenv3" argument is not provided.');
+        }
+
+        this._paymentInitializeOptions = adyenv3;
+
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(options.methodId);
+
+        this._adyenClient = await this._scriptLoader.load({
+            environment:  paymentMethod.initializationData.environment,
+            locale: this._locale,
+            clientKey: paymentMethod.initializationData.clientKey,
+            paymentMethodsResponse: paymentMethod.initializationData.paymentMethodsResponse,
+            showPayButton: false,
+        });
+
+        this._paymentComponent = await this._mountPaymentComponent(paymentMethod);
+
+        if (paymentMethod.method === AdyenPaymentMethodType.CreditCard ||
+            paymentMethod.method === AdyenPaymentMethodType.Bancontact) {
+            this._cardVerificationComponent = await this._mountCardVerificationComponent();
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment && payment.paymentData;
+        const shouldSaveInstrument = paymentData && (paymentData as HostedInstrument).shouldSaveInstrument;
```

Comment:
We are trying to limit usage of the keyword `as`, can we please use a type guard for this. There is already one present for HostedInstrument called `isHostedInstrumentLike`